DNN Platform/DotNetNuke.DependencyInjection/Extensions/BuildUpExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.DependencyInjection.Extensions
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+
+    /// <summary>
+    /// Adds property injection extension methods.
+    /// </summary>
+    public static class BuildUpExtensions
+    {
+        /// <summary>
+        /// Injects property dependency for public or proetected
+        /// properties that are decorated with <see cref="DependencyAttribute"/>.
+        /// </summary>
+        /// <param name="container">The service provider.</param>
+        /// <param name="myObject">The object to inject properties.</param>
+        public static void BuildUp(this IServiceProvider container, object myObject)
```

Comment:
Do we want this method to take an `ActionFilter`, to better document the expected usage?  We'd need to be explicit to create an overload if there was another type we wanted to support.

Reply:
I think that is a great idea. As far as I understand all WebAPI `ActionFilter` objects implement the `IFilter` interface. It should be safe to type check on that.

I would like to extend this code for MVC filters as well let's focus on WebAPI for now

Reply:
I really like the IFilter or similar limitation here.  Great idea @bdukes

Reply:
This has been updated in the latest code push to use `IFilter`. I'm going to resolve this conversation